apple/internal/framework_import_support.bzl

```diff
@@ -457,8 +460,30 @@ def _objc_provider_with_dependencies(
         objc_provider_fields["weak_sdk_framework"] = depset(weak_sdk_framework)
 
     objc_provider_fields.update(**additional_objc_provider_fields)
+    if not _OBJC_PROVIDER_LINKING:
+        objc_provider_fields = {"providers": additional_objc_providers}
+
     return apple_common.new_objc_provider(**objc_provider_fields)
 
+def _new_dynamic_framework_provider(*, cc_info, **kwargs):
+    """A wrapper API for the Bazel API of the same name to better support multiple Bazel versions
+
+    Args:
+        cc_info: CcInfo provider for the dynamic framework.
+        **kwargs: Additional arguments to pass if supported.
+    """
+
+    test_provider = apple_common.new_dynamic_framework_provider(
```

Comment:
Same as my comment in rules_swift, can we avoid creating a throw-away instance?

Reply:
done